examples/13-housing-nn-in-struct.rs

```diff
@@ -0,0 +1,234 @@
+//! This example is meant to show the following:
+//! 1. How to house a model in a struct.
+//! 2. How to seed a model for reproducibility.
+//! 3. Getting the shape of a runtime dimensioned tensor.
+//! 4. Dynamic batch size.
+
+// ************************************************************************************************
+// use
+// ************************************************************************************************
+
+use std::{cmp::min, time::Instant};
+
+use dfdx::{
+    data::{ExactSizeDataset, IteratorBatchExt, IteratorCollateExt, IteratorStackExt},
+    nn::builders::*,
+    optim::Adam,
+    prelude::{mse_loss, Optimizer},
+    shapes::{Const, HasShape, Rank1, Rank2},
+    tensor::{AsArray, AutoDevice, Gradients, SampleTensor, Tensor, TensorFrom, Trace},
+    tensor_ops::{Backward, BroadcastTo, RealizeTo, SelectTo, SumTo},
+};
+use indicatif::ProgressIterator;
+use rand::{rngs::StdRng, Rng, SeedableRng};
+
+// ************************************************************************************************
+// Model definition
+// ************************************************************************************************
+
+type Mlp = (
+    (Linear<2, 32>, ReLU),
+    (Linear<32, 32>, ReLU),
+    (Linear<32, 1>, Sigmoid),
+);
```

Comment:
Thoughts on adding a type alias for the built version?

```suggestion
type MlpStructure = (
    (Linear<2, 32>, ReLU),
    (Linear<32, 32>, ReLU),
    (Linear<32, 1>, Sigmoid),
);
type Mlp = <MlpStructure as BuildOnDevice<AutoDevice, f32>>::Built;
```

Reply:
Can you explain the reasoning behind this? what is the logical difference between MLP and MLPStructure ?